Makefile

```diff
@@ -73,7 +73,13 @@ endif
 include core/core.mk
 
 # Platform config is supposed to assign the targets
-ta-targets ?= user_ta
+ta-targets ?= invalid
+filtered-ta-targets := $(if $(CFG_USER_TA_TARGETS),$(filter $(CFG_USER_TA_TARGETS),$(ta-targets)),$(ta-targets))
+unsup-targets := $(filter-out $(filtered-ta-targets),$(CFG_USER_TA_TARGETS))
+ifneq (,$(unsup-targets))
+$(error CFG_USER_TA_TARGETS contains unsupported value(s): $(unsup-targets). Valid values: $(ta-targets))
+endif
+ta-targets := $(filtered-ta-targets)
```

Comment:
It's a bit of a hack to rewrite `ta-targets`. It would be more clean if the platform supplied a `supported-ta-targets`. From a quick look at the platforms it seems that we have the same pattern basic pattern again and again:
```
ta-targets = ta_arm32
ifeq ($(CFG_ARM64_core),y)
ta-targets += ta_arm64
endif
```
Perhaps we could have this in `core/arch/arm/arm.mk` instead:
```
# Provide default supported-ta-targets if not set by the platform config
ifneq (,$(supported-ta-targets))
supported-ta-targets = ta_arm32
ifeq ($(CFG_ARM64_core),y)
supported-ta-targets += ta_arm64
endif
endif
```

Reply:
@jenswi-linaro agreed. I will add a patch to do this first.